uber/models.py

```diff
@@ -593,27 +593,12 @@ def auto_badge_num(self, badge_type):
 
                 # Searches badge range for a gap in badge numbers; if none found, returns the latest badge number + 1
                 # Doing this lets admins manually set high badge numbers without filling up the badge type's range.
-                first = 0
-                last = sametype.count() - 1
-                if sametype_list[last] - sametype_list[first] == last:
+                start, end = c.BADGE_RANGES[badge_type][0], sametype_list[-1]
+                gap_nums = sorted(set(range(start, end + 1)).difference(sametype_list))
```

Comment:
Oh, and to address the last point, that... is actually a good point, and it would additionally assign multiple badges to the same number. Shoot.